### PR TITLE
Task/636 allow caching only specific subtypes to improve performance

### LINKED ISSUE #636: [TASK] only cache subtypes that are actually required

### PR DESCRIPTION
**Pull request by JoernBerkefeld**
# PR details

## What is the purpose of this pull request? (put an "X" next to an item)

- [x] Technical debt removed

## What changes did you make? (Give an overview)

- closes #636 

## blocked by
- Blocked by PR #631 

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] test scripts updated
- [ ] Wiki updated (if applicable)

### CODE REVIEW

File: lib/Retriever.js [hunk 1]
```diff
@@ -106,28 +110,28 @@ class Retriever {
...
                         );
                     }
                     Util.logger.info(
                         `Retrieving: ${metadataType}` +
                             (typeKeyMap[metadataType][0] === null
                                 ? ''
-                                : ` ${color.dim}(Keys: ${typeKeyMap[metadataType].join(', ')})${
-                                      color.reset
-                                  }`)
+                                : Util.getKeysString(typeKeyMap[metadataType]))
```
Comment (JoernBerkefeld) [inline lib/Retriever.js:124]: merge lines 122-124

File: lib/Deployer.js [hunk 2]
```diff
@@ -251,23 +251,29 @@ class Deployer {
...
+            );
             cache.setMetadata(type, result.metadata);
         }
         /** @type {TYPE.MultiMetadataTypeMap} */
         const multiMetadataTypeMap = {};
         // deploy metadata files, extending cache once deploys
-        for (const metadataType of deployOrder) {
+        for (const metadataType in deployOrder) {
             // TODO rewrite to allow deploying only a specific sub-type
             // const [type, subType] = metadataType.split('-');
```
Comment (DougMidgley) [inline lib/Deployer.js:275]: doubt check what deal is here? is this comment still needed?

File: lib/Retriever.js [hunk 3]
```diff
@@ -93,11 +96,12 @@ class Retriever {
                                 this.templateDir,
                                 name,
                                 templateVariables,
-                                subType
+                                subTypeArr?.[0] // this might cause issues if one were to try to use the method with multiple subtypes
```
Comment (DougMidgley) [inline lib/Retriever.js:99]: make sure comment is clear what objective is here

**Review by DougMidgley — CHANGES_REQUESTED** [review]

File: lib/Retriever.js (hunk 3 above)
Comment (JoernBerkefeld) [reply lib/Retriever.js:99]: added warning if more than 1 subtype was passed to retrieveAsTemplate: `retrieveAsTemplate only works with one subtype, ignoring all but first subtype from your list: ${subTypeArr.join(', ')`

**Comment by JoernBerkefeld** [comment]
merging pr after implementing requested changes

**Pull request merged**